File: fixtures/python/log_injection/positive.py
"""Positive fixtures for SEC015-log-injection (CWE-117)."""

import logging

logger = logging.getLogger(__name__)


# Case 1: logger.info with format placeholder + request param
def view(req):
    logger.info("user said {}".format(req.body))


# Case 2: logger.info with printf-style placeholder + request param
def handle(req):
    logger.info("user: %s", req.body)


# Case 3: logger.warning with request-style identifier
def process(request):
    logger.warning("processing: %s", request)


# Case 4: logging.debug with input identifier
def run(user_input):
    logging.debug("received: %s", user_input)

File: fixtures/rust/log_injection/positive.rs
// Positive fixtures for SEC015-log-injection (CWE-117)

use log;

struct Request;

// Case 1: log::info! with {} placeholder + param identifier
fn handler(req: Request) {
    log::info!("user: {}", req);
}

// Case 2: log::warn! with %s placeholder + request-style ident
fn process(request: Request) {
    log::warn!("processing: {}", request);
}

// Case 3: tracing::debug! with {} + param
fn trace_handler(input: String) {
    tracing::debug!("input: {}", input);
}

// Case 4: bare info! macro + request-style ident in request param
fn view(req: Request) {
    info!("received: {}", req);
}
